tests/e2escenarios/e2e_test.go

```diff
@@ -270,4 +302,110 @@ var _ = Describe("E2E Test", func() {
 			Eventually(string(commonVar.CliRunner.Run(getSVCArgs...).Out.Contents()), 60, 3).ShouldNot(ContainSubstring(serviceName))
 		})
 	})
+
+	Context("starting with non-empty Directory add Binding", func() {
+		componentName := helper.RandString(6)
+		var _ = BeforeEach(func() {
+			commonVar.CliRunner.EnsureOperatorIsInstalled("service-binding-operator")
+			commonVar.CliRunner.EnsureOperatorIsInstalled("cloud-native-postgresql")
+			Eventually(func() string {
+				out, _ := commonVar.CliRunner.GetBindableKinds()
+				return out
+			}, 120, 3).Should(ContainSubstring("Cluster"))
+			helper.Chdir(commonVar.Context)
+			helper.CopyExample(filepath.Join("source", "devfiles", "go"), commonVar.Context)
+			addBindableKind := commonVar.CliRunner.Run("apply", "-f", helper.GetExamplePath("source", "devfiles", "go", "cluster.yaml"))
+			Expect(addBindableKind.ExitCode()).To(BeEquivalentTo(0))
+		})
+
+		It("should verify developer workflow of using binding as env in innerloop", func() {
+			bindingName := helper.RandString(6)
+
+			command := []string{"odo", "init"}
+			_, err := helper.RunInteractive(command, nil, func(ctx helper.InteractiveContext) {
+
+				// helper.ExpectString(ctx, "Based on the files in the current directory odo detected")
+				helper.ExpectString(ctx, "Language: Go")
+				helper.ExpectString(ctx, "Project type: Go")
+				helper.ExpectString(ctx, "Is this correct")
+
+				helper.SendLine(ctx, "\n")
+
+				helper.ExpectString(ctx, "Select container for which you want to change configuration?")
+
+				helper.SendLine(ctx, "\n")
+
+				helper.ExpectString(ctx, "Enter component name")
+
+				helper.SendLine(ctx, componentName)
+
+				helper.ExpectString(ctx, "Your new component '"+componentName+"' is ready in the current directory")
+
+			})
+			Expect(err).To(BeNil())
+			Expect(helper.ListFilesInDir(commonVar.Context)).To(ContainElement("devfile.yaml"))
+
+			// // "execute odo dev and add changes to application"
+			var devSession helper.DevSession
+			var ports map[string]string
+
+			devSession, _, _, ports, err = helper.StartDevMode(nil)
+			Expect(err).ToNot(HaveOccurred())
+
+			// "send data"
+			recive_data := failcheck(ports["8080"])
+			Expect(recive_data).To(ContainSubstring("404 page not found")) // should fail as application is not connected to DB
```

Comment:
Since you are just calling an endpoint that always returns 404, this remains true even after the application is properly connected to the database.
How about calling an endpoint that needs to connect to the DB (like `/api/user`, which should actually fail until the application is bound to the DB)?

Reply:
So running this assertion in go routine? As we are adding the bindings later in the test. 
I think In a way we are testing this in this test case, isn't it? Testing if the application connects with the bindings after binding is added?

Reply:
I think just checking that a call to `/api/user` returns an error (because it requires the DB) before the binding is added. No goroutine needed.

You are already checking that the same call works after the binding is added  (via `rec := receiveData(fmt.Sprintf(ports["8080"] + "/api/user"))`)

Reply:
> I think In a way we are testing this in this test case, isn't it?

Yes, we are testing this but after the application is bound to the DB. I wanted to make sure that the error returned before the binding is added is really due to the application not bound to the DB, and not just because we are calling an invalid endpoint (which would return 404 at any time).